### DISCUSSION TITLE
ELSA for user-interaction-centric/forms-application

### DISCUSSION THREAD

**Discussion by Tobias-08** [elsa-workflows/elsa-core#2749] [Q&A]
Hi,

what would be the intended approach to use ELSA for a user-interaction-centric workflow application? This application has:

- Sequential or parallel tasks assigned to specific users (restrictive permissions)
- Forms to grab user input and task outcomes (approve, reject, ...)

Would this be the right approach?

- The application starts an ELSA workflow (e. g. via REST API), maybe providing workflow context
  - Is there a documentation for the REST API?
- The ELSA workflow creates tasks in the application (e. g. via custom activity) according to the defined process
- The application takes care of the form rendering and the permissions check
- Users open the task forms in the application und finish their tasks
- On finish the application informs the workflow (via signal or message?)
- The workflow continues creating tasks according to its process

Thanks!
Tobias

**Comment by sfmskywalker** [comment]
Hi Tobias,

That's 100% the right approach!

With regards to the REST API documentation, there's no documentation available from the documentation site as of yet, but you can enable the swagger UI from your ASP.NET app hosting Elsa.

And in case you want to look for the endpoints in code to try and infer what endpoints are available and what input they expect etc, take a look at [this folder](https://github.com/elsa-workflows/elsa-core/tree/master/src/server/Elsa.Server.Api/Endpoints). Each endpoint is implemented as an API controller.
